src/components/ReportingConfig/ReportingConfigForm.jsx

```diff
@@ -0,0 +1,512 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { ValidationFormGroup, Input, StatefulButton, Icon, Button } from '@edx/paragon';
+
+//  All the fields in this form that need to be validated can be added here.
+const REQUIRED_FIELDS = [
+  'hourOfDay',
+];
+// for the email delivery method
+const REQUIRED_EMAIL_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'emailRaw',
+];
+// for the sftp delivery mothod
+const REQUIRED_SFTP_FIELDS = [
+  ...REQUIRED_FIELDS,
+  'sftpPort',
+  'sftpHostname',
+  'sftpUsername',
+  'sftpPassword',
+  'sftpFilePath',
+];
+
+// for the submit button
+const SUBMIT_STATES = {
+  ERROR: 'error',
+  DEFAULT: 'default',
+  COMPLETE: 'complete',
+  PENDING: 'pending',
+};
+
+const MONTHLY_MAX = 31;
```

Comment:
This is probably outside the scope of this ticket, but do we actually support someone putting 31? Would they only get reports 7 times a year?

Reply:
This doesn't have to be researched/answered for this PR. Just something we'll want to check in the future.

Reply:
Good point, I'm not sure if anyone would want that. No easy solution here (that I can see), but I can put a note in the code if you'd like.